lib/eslint.js

```diff
@@ -476,27 +476,28 @@ module.exports = (function() {
     /**
      * Parses text into an AST. Moved out here because the try-catch prevents
      * optimization of functions, so it's best to keep the try-catch as isolated
-     * as possible
+     * as possible.
+     * This is equivalent to calling parser.parse with some exceptions:
+     *  - Errors are returned rather than thrown
+     *  - Empty input returns null
+     *  - Shebangs are ignored
      * @param {string} text The text to parse.
      * @param {Object} config The ESLint configuration object.
-     * @returns {ASTNode} The AST if successful or null if not.
+     * @returns {ASTNode|Error} The AST if successful or the error object if not.
      * @private
      */
-    function parse(text, config) {
+    api.parse = function(text, config) {
```

Comment:
By adding `.parse` as a method on `api`, are you attempting to address https://github.com/eslint/eslint/issues/948 as well?